src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts

```diff
@@ -0,0 +1,86 @@
+import { AlertService } from 'app/shared/alert/state/alert.service';
+import { LambdaEvent, LambdaEventsService } from 'app/shared/openapi';
+import { finalize } from 'rxjs/operators';
+
+/**
+ * Based on https://material.angular.io/components/table/examples example with expandable rows
+ * TODO: Filter by date (datasource is using timestamp instead of medium date)
+ * TODO: Change to prettier empty and error messages (cards)
+ * TODO: Friendly value map for reference (maybe success, maybe type too)
+ * TODO: Fix sort expanding every row
+ * TODO: Add pagination
+ * @export
+ * @class GithubAppsLogsComponent
+ * @implements {OnInit}
+ */
+@Component({
+  selector: 'app-github-apps-logs',
+  templateUrl: './github-apps-logs.component.html',
+  styleUrls: ['./github-apps-logs.component.scss'],
+  animations: [
+    trigger('detailExpand', [
+      state('collapsed', style({ height: '0px', minHeight: '0' })),
+      state('expanded', style({ height: '*' })),
+      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)'))
+    ])
+  ]
+})
+export class GithubAppsLogsComponent implements OnInit {
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: string,
+    private lambdaEventsService: LambdaEventsService,
+    private matSnackBar: MatSnackBar
+  ) {}
+  columnsToDisplay: string[] = ['repository', 'reference', 'success', 'type'];
+  displayedColumns: string[] = ['eventDate', 'githubUsername', ...this.columnsToDisplay];
+  lambdaEvents: LambdaEvent[];
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
+  loading = true;
+  public LambdaEvent = LambdaEvent;
+  dataSource: MatTableDataSource<LambdaEvent> = new MatTableDataSource();
+  expandedElement: LambdaEvent | null;
+  showContent: 'table' | 'error' | 'empty' | null;
+  isExpansionDetailRow = (i: number, row: Object) => row.hasOwnProperty('detailRow');
+
+  ngOnInit() {
+    this.loading = true;
+    this.dataSource.sort = this.sort;
+    this.lambdaEventsService
+      .getLambdaEventsByOrganization(this.data)
+      .pipe(
+        finalize(() => {
+          this.loading = false;
+          this.updateContentToShow(this.lambdaEvents);
+        })
+      )
+      .subscribe(
+        lambdaEvents => (this.lambdaEvents = lambdaEvents),
+        error => {
+          this.lambdaEvents = null;
+          const detailedErrorMessage = AlertService.getDetailedErrorMessage(error);
+          this.matSnackBar.open(detailedErrorMessage);
+        }
+      );
+  }
+
+  updateContentToShow(lambdaEvents: LambdaEvent[] | null) {
+    this.dataSource.data = lambdaEvents;
+    if (!lambdaEvents) {
+      this.showContent = 'error';
+    } else {
+      if (lambdaEvents.length === 0) {
+        this.showContent = 'empty';
+      } else {
+        this.showContent = 'table';
+      }
+    }
+  }
+
+  applyFilter(event: Event) {
+    const filterValue = (event.target as HTMLInputElement).value;
```

Comment:
Is there not a more Angular-ly way of doing this? Or is that covered by your TODO on filtering by dates.

Reply:
Seems fairly Angular to me I think https://angular.io/guide/user-input#get-user-input-from-the-event-object

Or did you have ngModel in mind?

Reply:
Given your link, it seems OK. You know Angular better than I, but it felt odd to me off-hand to be casting to a DOM element.